### PR TITLE
Clean up file generation

### PR DESCRIPTION
**Pull request by jtbandes**
- Removes the use of Bazel in favor of using `flatc` from the host machine. Pass e.g. `DOWNLOAD_FLATC=linux` as an env var to `generate_files.sh` to download it.
- Restructure so all generated files are in a `gen` folder (`src/gen` and `src/test/gen`)
- Update `generate_files.sh` to generate code from `flatbuffers.fbs` as well as test files

### CODE REVIEW

**Comment by jkuszmaul** [comment]
Getting reflection generated and moving the folders is great; I actively do *not* want to use flatc from the host machine---I want to make sure that this is using a specific agreed-upon version of flatbuffers. If upgrading the version within bazel is difficult, I'm happy to take care of it.

**Comment by jtbandes** [comment]
> I want to make sure that this is using a specific agreed-upon version of flatbuffers

how do you feel about the version check I added to the generation script?

**Comment by jkuszmaul** [comment]
> how do you feel about the version check I added to the generation script?

That would cover the primary concern, but at that point I'm also not entirely sure what the point is---it just enforces a particular version while making it so that whoever wants to modify this code has to go dig into the CI scripts to figure out how to run exactly the same thing that CI does (in fairness, "downloading & installing bazel" may be just as obnoxious as "downloading & installing flatc" for many, but for anyone already working with bazel I like to have things such that ideally any project _only_ requires that you have bazel installed for everything else to work. Obviously, that isn't true for the typescript side of this).

I could maybe be compelled to accept a patch which made `generate_files.sh` do the download + unzip + run flatc itself.

**Comment by jkuszmaul** [comment]
From offline conversation:

A `--download=PlatformName` to `generate_files.sh` would be fine with me so that there is an option where people don't have to locate and install the correct `flatc` themselves.

File: package.json [hunk 1]
```diff
@@ -8,7 +8,6 @@
     "url": "git+https://github.com/jkuszmaul/flatbuffers_reflection.git"
   },
   "author": "James Kuszmaul",
-  "license": "Apache-2.0",
```
Comment (jkuszmaul) [inline package.json:11]: Doesn't seem intentional? Unless `license` is not a supported tag in newer npm versions or something?

File: generate_files.sh [hunk 2]
```diff
@@ -1,26 +1,53 @@
...
+  mkdir -p tmp
+  curl -sL -o tmp/flatc.zip "$flatc_url"
+  unzip -o -d tmp tmp/flatc.zip
+  flatc_executable="tmp/flatc"
+fi
+
+"$flatc_executable" --version
+
+echo "Downloading reflection.fbs..."
+curl -sL -o src/reflection.fbs "https://raw.githubusercontent.com/google/flatbuffers/refs/tags/v$flatc_version/reflection/reflection.fbs"
```
Comment (jkuszmaul) [inline generate_files.sh:43]: Should this get put in `gen/`?

File: package.json (hunk 1 above)
Comment (jtbandes) [reply package.json:11]: there were two of these :)

File: generate_files.sh (hunk 2 above)
Comment (jtbandes) [reply generate_files.sh:43]: Technically it's not generated...? It is vendored though. I could move it into a vendor folder if you like? Also fine with putting it in gen if that's preferred.

Comment (jameskuszmaul-brt) [reply generate_files.sh:43]: A separate folder would be nice. Something to make clear that it is coming from elsewhere.

File: package.json (hunk 1 above)
Comment (jkuszmaul) [reply package.json:11]: Well it is Apache *2*.

**Review by jkuszmaul — APPROVED** [review]

**Pull request merged**